lib/src/dart2_9_suggestors/generated_factory_migrator.dart

```diff
@@ -25,37 +25,26 @@ import 'dart2_9_utilities.dart';
 class GeneratedFactoryMigrator extends RecursiveAstVisitor
     with AstVisitingSuggestorMixin
     implements Suggestor {
-  @override
-  visitArgumentList(ArgumentList node) {
-    super.visitArgumentList(node);
-
-    final generatedArg = getConnectGeneratedFactoryArg(node);
-    if (generatedArg == null) return;
-
-    final method = generatedArg.thisOrAncestorOfType<MethodInvocation>();
-    if (method != null && method.methodName.name == castFunctionName) {
-      return;
-    }
-
-    yieldPatch(generatedArg.offset, generatedArg.offset, '$castFunctionName(');
-    yieldPatch(generatedArg.end, generatedArg.end, ')');
-  }
-
   @override
   visitTopLevelVariableDeclaration(TopLevelVariableDeclaration node) {
     super.visitTopLevelVariableDeclaration(node);
 
     // Update only if the factory declaration is in the new boilerplate syntax.
-    if (isClassComponentFactory(node) && !isLegacyFactoryDecl(node)) {
-      final initializer = node.variables?.variables?.first?.initializer;
-      if (initializer is SimpleIdentifier) {
-        yieldPatch(
-            initializer.offset, initializer.offset, '$castFunctionName(');
-        yieldPatch(initializer.end, initializer.end, ')');
-      }
+    if (isClassOrConnectedComponentFactory(node) &&
+        !isLegacyFactoryDecl(node)) {
+      final generatedFactory = getGeneratedFactory(node);
+      if (generatedFactory
+              .thisOrAncestorOfType<MethodInvocation>()
```

Comment:
#nit This checks for any ancestor, but we might want to make sure it's wrapped directly around the generated factory:

```suggestion
      final generatedFactory = getGeneratedFactory(node);
      final parent = generatedFactory.parent;
      if (parent is MethodInvocation && parent
```